### PR TITLE
Moved 'Making Good PRs' section up in the hierarchy and clarified its content

### PR DESCRIPTION
**Pull request by swastim01**
### Description  
- Enforced a stricter PR title format to improve clarity and maintainability.  
- Encouraged writing accurate and concise PR titles/descriptions.  
- Suggested promoting the **"Making Good PRs"** section by moving it up in the documentation hierarchy for better visibility.  

These changes aim to enhance the quality of PR titles, making historical searches more effective while improving documentation organization.

### CODE REVIEW

**Comment by hugovk** [comment]
It's hard to review the changes because the section was moved in the page, please could you move it back?

I think it makes sense to keep the original order, where the "quick guide" comes first.

**Comment by swastim01** [comment]
Hi, I have done the needful. Please let me know if any other changes are required.

File: getting-started/pull-request-lifecycle.rst [hunk 1]
```diff
@@ -183,59 +183,93 @@ message. It is usually okay to leave that as-is and close the editor.
 See `the merge command's documentation <https://git-scm.com/docs/git-merge>`_
 for a detailed technical explanation.
 
-
 .. _good-prs:
 
-Making good PRs
+Making Good PRs
```
Comment (hugovk) [inline getting-started/pull-request-lifecycle.rst:188]: Please retain sentence case: https://devguide.python.org/documentation/style-guide/#capitalization
```suggestion
Making good PRs
```
Same applies to the headers below.

File: getting-started/pull-request-lifecycle.rst [hunk 2]
```diff
@@ -183,59 +183,93 @@ message. It is usually okay to leave that as-is and close the editor.
...
-#. **Make sure all tests pass.** The entire test suite needs to
-   :ref:`run <runtests>` **without failure** because of your changes.
-   It is not sufficient to only run whichever test seems impacted by your
-   changes, because there might be interferences unknown to you between your
-   changes and some other part of the interpreter.
-
-#. Proper :ref:`documentation <documenting>` additions/changes should be included.
+When creating a pull request, following best practices ensures your contribution is **clear, maintainable, and easy to review**. A well-structured PR improves collaboration and speeds up the review process.
+
+1. **Use a Clear and Structured PR Title**
```
Comment (hugovk) [inline getting-started/pull-request-lifecycle.rst:193]: Can we keep using `#.` so Sphinx auto-numbers, and we don't need to hardcode?

```suggestion
#. **Use a clear and structured PR title**
```

Also please trim trailing spaces, or install pre-commit to do it for you:

https://devguide.python.org/getting-started/setup-building/#install-pre-commit

File: getting-started/pull-request-lifecycle.rst [hunk 3]
```diff
@@ -183,59 +183,93 @@ message. It is usually okay to leave that as-is and close the editor.
...
-   changes and some other part of the interpreter.
-
-#. Proper :ref:`documentation <documenting>` additions/changes should be included.
+When creating a pull request, following best practices ensures your contribution is **clear, maintainable, and easy to review**. A well-structured PR improves collaboration and speeds up the review process.
+
+1. **Use a Clear and Structured PR Title**  
+
+   PR titles often become commit messages, making them **critical for maintainability and searchability**. Follow these guidelines:
+
+    **Do:**
```
Comment (hugovk) [inline getting-started/pull-request-lifecycle.rst:197]: See the preview: https://cpython-devguide--1510.org.readthedocs.build/getting-started/pull-request-lifecycle/#making-good-prs

Don't blockquote this. Same for "Avoid".

File: getting-started/pull-request-lifecycle.rst [hunk 4]
```diff
@@ -183,59 +183,93 @@ message. It is usually okay to leave that as-is and close the editor.
...
-   ``main``.
-
-#. **Make sure to follow Python's style guidelines.** For Python code you
-   should follow :PEP:`8`, and for C code you should follow :PEP:`7`. If you have
-   one or two discrepancies those can be fixed by the core developer who merges
-   your pull request. But if you have systematic deviations from the style guides
-   your pull request will be put on hold until you fix the formatting issues.
+When creating a pull request, following best practices ensures your contribution is **clear, maintainable, and easy to review**. A well-structured PR improves collaboration and speeds up the review process.
+
+1. **Use a Clear and Structured PR Title**
```
Comment (JelleZijlstra) [inline getting-started/pull-request-lifecycle.rst:193]: This is general guidance, not specific to Python. I'm not sure it makes sense to give this sort of general guideline: the devguide is more helpful to developers who want to start contributing to Python if it talks specifically about what's special about this project.

File: getting-started/pull-request-lifecycle.rst [hunk 5]
```diff
@@ -183,59 +183,93 @@ message. It is usually okay to leave that as-is and close the editor.
...
+   Think about how your change affects existing users before submitting your PR.
+
+6. **Ensure Proper Testing**
+
+   - Every PR should include **appropriate test cases** to validate the changes.
+   - PRs without tests **will not be accepted** unless they are purely documentation changes.
+   - Tests should **cover edge cases** and expected behaviors.
+   - For bug fixes, add a test that **fails without the fix** and **passes after applying it**.
+
+7. **Make Sure All Tests Pass**
```
Comment (StanFromIreland) [inline getting-started/pull-request-lifecycle.rst:258]:
```suggestion
#. **Ensure all tests pass**
```
